src/libraries/System.Private.CoreLib/src/System/Environment.cs

```diff
@@ -18,6 +18,11 @@ public static partial class Environment
         /// </summary>
         internal static bool IsSingleProcessor => ProcessorCount == 1;
 
+        /// <summary>
+        /// Gets whether the current process is authorized to perform security-relevant functions.
+        /// </summary>
+        public static bool IsPrivilegedProcess { get; } = IsAdminProcess();
```

Comment:
This is not pay-for-play. `{ get; } =` will cause the `IsAdminProcess` check to be executed during `Environment` static constructor that runs in early in nearly all apps (e.g. the Environment static constructor is triggered by `Environment.ProcessorCount`.).

This should be done lazily and in trim-friendly fashion. Check `Environment.ProcessPath` or `enum TriState` for some ideas.

Reply:
Since it is for one time initialization I guess we can use `ProcessId` pattern like: 
```c#
        private static bool s_privilegedProcess;
        private static volatile int s_pendingPrivilegedProcess;

        /// <summary>
        /// Gets whether the current process is authorized to perform security-relevant functions.
        /// </summary>
        public static bool IsPrivilegedProcess
        {
            get
            {
                if (s_pendingPrivilegedProcess == 0)
                {
                    s_privilegedProcess = IsAdminProcess();
                    s_pendingPrivilegedProcess = 1;
                }
                return s_privilegedProcess;
            }
        }
```